### PR TITLE
Add sample dashboard for Cloud Spanner

### PR DESCRIPTION
**Pull request by nielm**
A simple sample dashboard for Cloud Spanner with a number of useful performance monitoring graphs.

### CODE REVIEW

**Comment by nielm** [comment]
R: @EvanSimpson

File: dashboards/spanner/spanner.json [hunk 1]
```diff
@@ -0,0 +1,499 @@
...
+                    "aggregation": {
+                      "alignmentPeriod": "60s",
+                      "crossSeriesReducer": "REDUCE_SUM",
+                      "groupByFields": [
+                        "metric.label.\"database\"",
+                        "resource.label.\"instance_id\""
+                      ],
+                      "perSeriesAligner": "ALIGN_MEAN"
+                    },
+                    "filter": "metric.type=\"spanner.googleapis.com/instance/storage/used_bytes\" resource.type=\"spanner_instance\" resource.label.\"instance_id\"=\"beam-test\" resource.label.\"project_id\"=\"spannertest-352708\"",
```
Comment (EvanSimpson) [inline dashboards/spanner/spanner.json:348]: Remove instance and project id here as well.

File: dashboards/spanner/spanner.json [hunk 2]
```diff
@@ -0,0 +1,499 @@
...
+                "legendTemplate": "Processing units",
+                "plotType": "LINE",
+                "targetAxis": "Y1",
+                "timeSeriesQuery": {
+                  "timeSeriesFilter": {
+                    "aggregation": {
+                      "crossSeriesReducer": "REDUCE_MAX",
+                      "perSeriesAligner": "ALIGN_MEAN"
+                    },
+                    "filter": "metric.type=\"spanner.googleapis.com/instance/processing_units\" resource.label.\"instance_id\"=\"beam-test\" resource.label.\"project_id\"=\"spannertest-352708\" resource.type=\"spanner_instance\""
```
Comment (EvanSimpson) [inline dashboards/spanner/spanner.json:305]: Remove instance and project id

File: dashboards/spanner/spanner.json [hunk 3]
```diff
@@ -0,0 +1,499 @@
...
+                "plotType": "HEATMAP",
+                "targetAxis": "Y1",
+                "timeSeriesQuery": {
+                  "timeSeriesFilter": {
+                    "aggregation": {
+                      "alignmentPeriod": "60s",
+                      "crossSeriesReducer": "REDUCE_SUM",
+                      "perSeriesAligner": "ALIGN_DELTA"
+                    },
+                    "filter": "metric.type=\"spanner.googleapis.com/api/request_latencies\" resource.type=\"spanner_instance\" metric.label.\"method\"=monitoring.regex.full_match(\"ExecuteBatchDml|ExecuteSql|ExecuteStreamingSql|Read|StreamingRead\")"
```
Comment (EvanSimpson) [inline dashboards/spanner/spanner.json:117]: Are each of the "Read" charts displaying one line per method label + instance? Would users with many instances benefit from all of the labels being aggregated to a single line per instance, or is it more important to see which individual methods have higher latencies?

**Review by EvanSimpson — CHANGES_REQUESTED** [review]

File: dashboards/spanner/spanner.json (hunk 1 above)
Comment (nielm) [reply dashboards/spanner/spanner.json]: Oops - thanks, done

File: dashboards/spanner/spanner.json (hunk 2 above)
Comment (nielm) [reply dashboards/spanner/spanner.json]: Oops - thanks, done

File: dashboards/spanner/spanner.json (hunk 3 above)
Comment (nielm) [reply dashboards/spanner/spanner.json:117]: These are all the possible read methods that can be used, and yes they are displaying a single line for all methods. 

These graphs are effectively the same as in the Cloud Spanner monitoring page in the cloud console (except with Read and Write split into separate graphs). 

If the end user wants to dig further into the metrics, then they can always go into the metrics explorer for these graphs, and filter by method.

**Pull request merged**